src/lib.rs

```diff
@@ -121,11 +123,14 @@ pub fn edit_distance(search_term: &str, known_term: &str) -> usize {
 
     // Run the algorithm.
     for i in 1..n {
+        let mut known_chars = known_term.chars();
```

Comment:
Additionally, I believe this line should be run outside of the loop. It remains the same regardless of loop iteration and is thus being run unnecessarily `n - 3` times more than it should be.

Reply:
Yeah you are right, But actually, the `collection` allocation has far more overhead than extra iteration.

Reply:
That's interesting. I agree with replacing it with the allocation you have here. One thing is that I'm not sure that it's doing that same thing since the `.next()` allocation seems to be indexing at `i ` whereas the old implementation is indexing at `i - 1`. We also need the ability to index at `i - 2` for the transposition calculation